### DISCUSSION TITLE
Document hard-coded implicit delay in Node.js

### DISCUSSION THREAD

**Discussion by jlissner** [mswjs/mswjs.io#376] [Q&A] [answered]
- [Open page](https://mswjs.io/docs/api/delay/)

## Feedback

It says that it waits `~100-400ms` by default, but that is not always the case. From the [source code](https://github.com/mswjs/msw/blob/main/src/core/delay.ts):

```js
function getRealisticResponseTime(): number {
  if (isNodeProcess()) {
    return NODE_SERVER_RESPONSE_TIME
  }

  return Math.floor(
    Math.random() * (MAX_SERVER_RESPONSE_TIME - MIN_SERVER_RESPONSE_TIME) +
      MIN_SERVER_RESPONSE_TIME,
  )
}
```

So if it's determined to be a node process, it always returns `5`. I'm using `msw` for testing a react app, so while it's running on a node process, that is not the desired behavior. Additionally, the docs make no mention of this "feature", which I had to dig into the source code to understand why thing's weren't being delayed as expected.

**Comment by kettanaito** [comment, marked as answer]
Hi, @jlissner. Thanks for bringing this up.

Yes, that's indeed an implicit feature designed to provide a better testing experience since in the light of how the realistic server response time is used with the `delay()` function. 

MSW intends that use case to enable server-like response times when developing, but when testing you want your network to be fast. That's why we preemptively hard-code the realistic response time to a fixed value when run in Node.js. Your tests should be reliable and reproducible, which means a reliable and reproducible network that MSW aims to provide, including the fixed response timing when using the `delay()` function without any arguments.

You can opt-out from this by providing an explicit delay duration to the `delay()` function. In that case, MSW will respect it as-is. 

I do believe this behavior makes sense, we just need to document it better on the website.

**Reply by jlissner** [reply]
Thanks for the answer! I think a nice-to-have would be to explicitly state if you want `browser` or `server` default/`real` delays. That being said, just having it documents would be great though :).
